### PR TITLE
SONARJAVA-1153 Fix false positive in printfCheck

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/PrintfCheck.java [hunk 1]
```diff
@@ -107,11 +109,15 @@ protected void onMethodInvocationFound(MethodInvocationTree mit) {
         return;
       }
       verifyParameters(mit, args, params);
-    } else if (formatStringTree.is(Tree.Kind.PLUS)) {
+    } else if (formatStringTree.is(Tree.Kind.PLUS) && operandsAreOnSameLine((BinaryExpressionTree) formatStringTree)) {
       addIssue(mit, "Format specifiers should be used instead of string concatenation.");
     }
   }
 
+  private boolean operandsAreOnSameLine(BinaryExpressionTree formatStringTree) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/PrintfCheck.java]: MAJOR Make "operandsAreOnSameLine" a "static" method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [The Cyclomatic Complexity of this method "onMethodInvocationFound" is 11 which is greater than 10 authorized.](https://github.com/SonarSource/sonar-java/blob/64e82ce01b1fe4b27e52186770764f53d7895689/java-checks/src/main/java/org/sonar/java/checks/PrintfCheck.java#L81) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-checks/src/main/java/org/sonar/java/checks/PrintfCheck.java [hunk 2]
```diff
@@ -107,11 +109,15 @@ protected void onMethodInvocationFound(MethodInvocationTree mit) {
         return;
       }
       verifyParameters(mit, args, params);
-    } else if (formatStringTree.is(Tree.Kind.PLUS)) {
+    } else if (formatStringTree.is(Tree.Kind.PLUS) && operandsAreOnSameLine((BinaryExpressionTree) formatStringTree)) {
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/PrintfCheck.java]: With such implementation we will drop reported False Positives, but we won't be able to detect issue such as:
```java
String.format("value is %d"
      + " not replaced"); // False negative, "%d" is not replaced
```

**Comment by m-g-sonar** [comment]
We won't be really more precise, but we will reduce number of FP cause by line breaks, so LGTM!

**Pull request merged**